hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -315,23 +317,56 @@ def post_apply(self, updated: t.Dict = {}):
         table = dynamodb.Table(self.data_store_table)
 
         for update in updated.values():
-            row = update.as_csv_row()
+            row: t.List[str] = update.as_csv_row()
             # example row format: ('<raw_indicator>', '<indicator-id>', '<descriptor-id>', '<time added>', '<space-separated-tags>')
             # e.g (10736405276340','096a6f9...064f', '1234567890', '2020-07-31T18:47:45+0000', 'true_positive hma_test')
-            if PDQSignalMetadata(
-                signal_id=int(row[1]),
+            new_tags = row[4].split(" ") if row[4] else []
+
+            # Figure out if we have a new opinion about this indicator and clear out a pending change if so
+            # If this is a new indicator without metadata we also should have no pending opinion about it
+            new_pending_opinion_change = PendingOpinionChange.NONE
+            for signal in PDQSignalMetadata.get_from_signal(
+                table, int(row[1]), S3ThreatDataConfig.SOURCE_STR
+            ):
+                # Find the metadata object for this privacy group if it exists
+                if signal.ds_id == str(self.privacy_group):
+
+                    # Threat Updates guarentees there is either 0 or 1 special tags on a descriptor
+                    opinion_tags = ThreatDescriptor.SPECIAL_TAGS
+                    old_opinion = [tag for tag in signal.tags if tag in opinion_tags]
+                    new_opinion = [tag for tag in new_tags if tag in opinion_tags]
+
+                    # If our opinion changed, set the pending opinion change to None
+                    # otherwise keep it unchanged
+                    new_pending_opinion_change = (
```

Comment:
nit: wondering if it is worth checking for the case where it hasn't changed but it is the pending state already (I think it is possible if we fetch right when a change is requested on the UI but that is pretty niche)